classes/class-extensions.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace HM\BackUpWordPress;
+
+/**
+ * Class Extensions
+ * @package HM\BackUpWordPress
+ */
+final class Extensions {
+
+	/**
+	 * Contains the instantiated Extensions instance.
+	 *
+	 * @var Extensions $this->instance
+	 */
+	private static $instance;
+
+	/**
+	 * Holds the root URL of the API.
+	 *
+	 * @var string
+	 */
+	protected $root_url = '';
+
+	/**
+	 * Extensions constructor.
+	 *
+	 */
+	private function __construct() {
+
+		$this->root_url  = 'https://bwp.hmn.md/wp-json/wp/v2/';
+
+	}
+
+	private function __wakeup() {}
+
+	private function __clone() {}
+
+	/**
+	 * Returns the *Singleton* instance of this class.
+	 *
+	 * @staticvar Extensions $instance The *Singleton* instances of this class.
+	 *
+	 * @return Extensions The *Singleton* instance.
+	 */
+	public static function get_instance() {
+
+		if ( ! ( self::$instance instanceof Extensions ) ) {
+
+			self::$instance = new Extensions();
+
+		}
+
+		return self::$instance;
+
+	}
+
+	/**
+	 * Parses the body of the API response and returns it.
+	 *
+	 * @return array|bool|mixed|object
+	 */
+	public function get_edd_data() {
+
+		$response = $this->fetch( 'edd-downloads', 600 );
```

Comment:
Is the `600` ttl intentional here? It's overriding the default of `DAY_IN_SECONDS`.